### PR TITLE
Adds literacy check to reading and writing papers.

### LINKED ISSUE #12277: Drones can write on paper but can't read it.
Since monkeys communicating with labelers was a bug, drones writing on paper is likely one as well.

#ChainedDrone

### LINKED ISSUE #17561: Devils in the third form cannot read
Ironic as fuck
that entire form is just a downgrade tbh

### PR DESCRIPTION
**Pull request by lordpidey**
Fixes #12277
Fixes #17561

Small fix, adds a is_literate proc to mobs.  I'll make the proc more fleshed out in another PR.

### CODE REVIEW

File: code/game/gamemodes/demon/true_demon/_true_demon.dm [hunk 1]
```diff
@@ -90,13 +90,11 @@
 /mob/living/carbon/true_devil/IsAdvancedToolUser()
 	return 1
 
-
 /mob/living/carbon/true_devil/canUseTopic()
 	if(stat)
 		return
 	return 1
 
-
```
Comment (Supermichael777) [inline code/game/gamemodes/demon/true_demon/_true_demon.dm]: line spacings between procs are for readability.  The segmentation helps readers determen where procs are.

IIRC this has no effect on the compiled code so please don't remove it

Comment (lordpidey) [inline code/game/gamemodes/demon/true_demon/_true_demon.dm]: The spacing was inconsistent within the file, and has been made consistent.  

That said, if you think I should change all of them to be two lines between procs, that's also possible.

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -75,17 +75,15 @@
...
-		return
-
-	if(!ishuman(usr))
-		return
-	var/mob/living/carbon/human/H = usr
-	if(H.disabilities & CLUMSY && prob(25))
-		H << "<span class='warning'>You cut yourself on the paper! Ahhhh! Ahhhhh!</span>"
-		H.damageoverlaytemp = 9001
-		H.update_damage_hud()
+	if(usr.stat || !usr.canmove || usr.restrained()|| !usr.is_literate())
```
Comment (phil235) [inline code/modules/paperwork/paper.dm]: replace those three checks by usr.incapacitated()

File: code/game/gamemodes/demon/true_demon/_true_demon.dm (hunk 1 above)
Comment (Supermichael777) [inline code/game/gamemodes/demon/true_demon/_true_demon.dm]: oh i didnt see it was two yeah one is fine

**Comment by AnturK** [comment]
The simple human mobs probably should be literate but i guess that can be changed if needed.

**Pull request merged**